### PR TITLE
feat(rust): add ockam_error

### PR DESCRIPTION
**Pull request by SanjoDeundiak**

### CODE REVIEW

**Review by mrinalwadhwa — APPROVED** [review]

File: implementations/rust/.gitignore [hunk 1]
```diff
@@ -1,2 +1,4 @@
 # Cargo stores the output of a build into the target directory.
 target
+/Cargo.toml
+/Cargo.lock
```
Comment (mrinalwadhwa) [inline implementations/rust/.gitignore:4]: could you move this change to a separate commit.

Comment (SanjoDeundiak) [reply implementations/rust/.gitignore:4]: @mrinalwadhwa fixed
